kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java

```diff
@@ -631,5 +634,36 @@ private InternalClusterMetrics calculateClusterMetrics(
         .map(ac -> ac.deleteRecords(records)).then();
   }
 
+  @SneakyThrows
+  private Mono<String> increaseTopicPartitions(AdminClient adminClient,
+                                               Map<String, NewPartitions> newPartitionsMap,
+                                               String topicName) {
+    return ClusterUtil.toMono(adminClient.createPartitions(newPartitionsMap).all(), topicName);
+  }
+
+  @SneakyThrows
+  public Mono<PartitionsIncreaseResponse> increaseTopicPartitions(AdminClient adminClient,
+                                               Mono<PartitionsIncrease> partitionsIncrease) {
+    return partitionsIncrease.flatMap(
+        partitions -> {
+          Map<String, NewPartitions> newPartitionsMap = Collections.singletonMap(
+              partitions.getTopicName(),
+              NewPartitions.increaseTo(partitions.getTotalPartitionsCount())
+          );
+          return increaseTopicPartitions(adminClient, newPartitionsMap, partitions.getTopicName())
+              .flatMap(
+                  topicName -> getTopicsData(adminClient, Collections.singleton(topicName))
+                      .map(t -> new PartitionsIncreaseResponse()
+                          .topicName(topicName)
+                          .totalPartitionsCount(t.getPartitionCount()))
+                      .next());
+        });
+  }
+
+  public Mono<PartitionsIncreaseResponse> increaseTopicPartitions(KafkaCluster cluster,
+                                                     Mono<PartitionsIncrease> partitionsIncrease) {
+    return getOrCreateAdminClient(cluster)
+        .flatMap(ac -> increaseTopicPartitions(ac.getAdminClient(), partitionsIncrease));
```

Comment:
looks strange that you pass mono to function, you can unwrap it here

Reply:
How to do it better? Use block() or nested flatMap() or something else?

Reply:
nested flatmap